### PR TITLE
Add a Check method overload to take Formula as input

### PR DESCRIPTION
**Pull request by Robulane**
This PR adds a simple method overload that takes a Formula instead of a string in RecalcEngine.

Motivation: Thanks to this change, we can re-use already parsed formula that will lead to significant performance improvements.

### CODE REVIEW

**Comment by MikeStall** [comment]
fyi, you'll want to rebase on latest to get:
[https://github.com/microsoft/Power-Fx/commit/c38806c836960153c23076ad8561121d9ee8e388](https://github.com/microsoft/Power-Fx/commit/c38806c836960153c23076ad8561121d9ee8e388)

File: src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs [hunk 1]
```diff
@@ -136,15 +136,30 @@ public CheckResult Check(string expressionText, FormulaType parameterType = null
...
         }
 
+        /// <summary>
+        /// Type check a formula without executing it. 
+        /// </summary>
+        /// <param name="formula">The formula to check.</param>
+        /// <param name="parameterType">The type context.</param>
+        /// <remarks>Used by PVA.</remarks>
+        /// <returns></returns>
+        internal CheckResult Check(Formula formula, FormulaType parameterType = null)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs:146]: >Formula [](http://example.com/codeflow?start=35&length=7)

Note that we're tyring to move off internals, so Formula would need to be public too.

Comment (Robulane) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: I tried, it changes a lot of things from internal to public and starts to require the summaries...

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: Should we go for a 2 stage approach and keep the Formula object internal?

public ParseResult {
     public TexlNode ParseTree => Formula.ParseTree;
     public ExpressionError[] Errors => Formula.GetParseErrors();
     public bool IsSuccess => Errors == null;


     private Formula _formula;
}

ParseResult Parse(string expression)

CheckResult Check(IParseResult parseResult, FormulaType parameterType = null)

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: obviously also still keeping 
CheckResult Check(string expression, FormulaType parameterType)

Comment (Robulane) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: TexlNode is also internal that brings a lot of other internal stuff. Let me see if I can give this PR to the team to follow up change internal to public

Comment (ivanradicek) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: FYI, `TexlNode` should be public soon. See: #248.

Comment (Robulane) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: Oh, nice, then I'll wait for this PR to be merged :)

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs]: We do need to huddle about the right shape for the public surface here.  Ability to parse-only makes sense, but Formula class may not be the right public surface.

**Comment by MikeStall** [comment]
Superceded by #319, which has been merged.

**Pull request closed without merging**